test/com/google/enterprise/adaptor/ValidatedUriTest.java

```diff
@@ -26,69 +29,91 @@
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("");
   }
 
   @Test
   public void testNoProtocol() throws Exception {
-    thrown.expect(InvalidConfigurationException.class);
-    urlValidator.validate("//foo/bar");
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("//foo/bar");
   }
 
   @Test
   public void testUnknownProtocol() throws Exception {
-    thrown.expect(InvalidConfigurationException.class);
-    urlValidator.validate("unknown://foo/bar");
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("unknown://foo/bar");
   }
 
   @Test
   public void testBadProtocol() throws Exception {
-    thrown.expect(InvalidConfigurationException.class);
-    urlValidator.validate("https//foo/bar");
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("https//foo/bar");
   }
 
   @Test
   public void testNoHost() throws Exception {
-    thrown.expect(InvalidConfigurationException.class);
-    urlValidator.validate("http://");
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("http://");
+  }
+
+  @Test
+  public void testNoPath() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("http://foo:80");
+  }
+
+  @Test
+  public void testRootPath() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("http://foo:80/");
+  }
+
+  @Test
+  public void testRelativeUri() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("foo/bar");
   }
 
   @Test
   public void testMessageFormatRemnants() throws Exception {
-    thrown.expect(InvalidConfigurationException.class);
-    urlValidator.validate("http://message_format/foo/{0}");
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("http://message_format/foo/{0}");
   }
 
   @Test
   public void testNakedIPv6Address() throws Exception {
-    thrown.expect(InvalidConfigurationException.class);
-    urlValidator.validate("http://::1/foo/bar");
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("http://::1/foo/bar");
   }
 
   @Test
   public void testBracketedIPv6Address() throws Exception {
-    assertEquals(true, urlValidator.validate("http://[::1]/foo/bar"));
+    assertEquals(new URI("http://[::1]/foo/bar"),
+        new ValidatedUri("http://[::1]/foo/bar").getUri());
+  }
+
+  @Test
+  public void testGetUri() throws Exception {
+    assertEquals(new URI("http://example.com/foo/bar"),
+        new ValidatedUri("http://example.com/foo/bar").getUri());
   }
 
   @Test
   public void testReachableHost() throws Exception {
-   assertEquals(true, urlValidator.validate("http://127.0.0.1/foo/bar"));
+    new ValidatedUri("http://127.0.0.1/foo/bar").testHostIsReachable();
   }
 
   @Test
   public void testUnreachableHost() throws Exception {
-   assertEquals(false, urlValidator.validate("http://unknown_host/foo/bar"));
+    new ValidatedUri("http://unknown-host/foo/bar").testHostIsReachable();
```

Comment:
Holy network access in a unit test, Batman! Should be quick, so OK, but I think it's worth grabbing Thomas's log machinery to verify the log messages. See Logging.java in the documentum (v4) tests or the CM prod util/testing package.